mongo/pyproject.toml

```diff
@@ -37,7 +37,7 @@ dynamic = [
 
 [project.optional-dependencies]
 deps = [
-    "pymongo[srv]==3.12.3",
+    "pymongo[srv]==4.2; python_version >= '3.7'"
```

Comment:
Should we keep `pymongo[srv]==3.12.3` for `python_version`s < 3.7?

Reply:
My understanding is that there are too many differences in the library's API so keeping code compatibility with both is not something we'd like to do.

Reply:
[Mongo check will only be installed with python3](https://github.com/DataDog/integrations-core/pull/12594/commits/1eb5793df5a1ec429cc575eb39440d8389957b26)